src/iot/commands/editPolicyVersion.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { getLogger } from '../../shared/logger'
+import { getTabSizeSetting } from '../../shared/utilities/editorUtilities'
+import { IotPolicyVersionNode } from '../explorer/iotPolicyVersionNode'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+
+export async function editPolicyVersionCommand(node: IotPolicyVersionNode) {
+    getLogger().debug('EditPolicyVersion called for %O', node)
+
+    const policyVersionId = node.version.versionId!
+    const policyName = node.policy.name
+
+    try {
+        const policy = await node.iot.getPolicyVersion({ policyName, policyVersionId })
+        const document = policy.policyDocument
+
+        if (!document) {
+            return
```

Comment:
A policy must have a policy document, so if `getPolicyVersion` succeeds then this should be defined. Since `GetPolicyVersionResponse.policyDocument` is defined as a `string | undefined`, I used this so the compiler would know that this is not null. Should I use the `policy.policyDocument!` assertion operator instead?

Reply:
If the `return` should never be hit, go ahead and assert with the exclamation mark (if you're using `document` in more than one place, this could be OK  since it means you won't have to assert every time. Just make a note that the return won't be hit, either in comment or error message form.